chalicelib/checks/higlass_checks.py

```diff
@@ -0,0 +1,1241 @@
+
+            new_viewconfs[title] = post_viewconf_results["view_config_uuid"]
+
+        # The other_processed_files section has been updated. Patch the changes.
+        try:
+            # Make sure all higlass_view_config fields just show the uuid.
+            for g in expsets_to_update[accession]:
+                if isinstance(g["higlass_view_config"], dict):
+                    uuid = g["higlass_view_config"]["uuid"]
+                    g["higlass_view_config"] = uuid
+
+            ff_utils.patch_metadata(
+                {'other_processed_files': expsets_to_update[accession]},
+                obj_id=accession,
+                key=connection.ff_keys,
+                ff_env=connection.ff_env
+            )
+        except Exception as e:
+            if accession not in action_logs['failed_to_patch_expset']:
+                action_logs['failed_to_patch_expset'][accession] = {}
+            if title not in action_logs['failed_to_patch_expset'][accession]:
+                action_logs['failed_to_patch_expset'][accession][title] = {}
+            action_logs['failed_to_patch_expset'][accession][title] = str(e)
+            continue
+
+        # Success. Note which titles link to which HiGlass view configs.
+        if accession not in action_logs['successes']:
+            action_logs['successes'][accession] = {}
+        action_logs['successes'][accession] = new_viewconfs
+
+    action.status = 'DONE'
+    action.output = action_logs
+    return action
+
+@check_function(confirm_on_higlass=False, filetype='all', higlass_server=None)
+def files_not_registered_with_higlass(connection, **kwargs):
+    """
+    Used to check registration of files on higlass and also register them
+    through the patch_file_higlass_uid action.
+
+    If confirm_on_higlass is True, check each file by making a request to the
+    higlass server. Otherwise, just look to see if a higlass_uid is present in
+    the metadata.
+
+    The filetype arg allows you to specify which filetypes to operate on.
+    Must be one of: 'all', 'bigbed', 'mcool', 'bg', 'bw', 'beddb', 'chromsizes'.
+    'chromsizes' and 'beddb' are from the raw files bucket; all other filetypes
+    are from the processed files bucket.
+
+    higlass_server may be passed in if you want to use a server other than
+    higlass.4dnucleome.org.
+
+    Since 'chromsizes' file defines the coordSystem (assembly) used to register
+    other files in higlass, these go first.
+
+    Args:
+        connection: The connection to Fourfront.
+        **kwargs
+
+    Returns:
+        A check/action object.
+    """
+    check = init_check_res(connection, 'files_not_registered_with_higlass')
+    check.status = "FAIL"
+    check.description = "not able to get data from fourfront"
+    # keep track of mcool, bg, and bw files separately
+    valid_filetypes = {
+        "raw": ['chromsizes', 'beddb'],
+        "proc": ['mcool', 'bg', 'bw', 'bed', 'bigbed'],
+    }
```

Comment:
This is minor, but the same information is essentially contained by `valid_filetypes` and `valid_types_raw`/`valid_types_proc`, causing us to have to change multiple spots if we want to add more filetypes. Could you use one or the other? Seems like the only spot the new `valid_filetypes` is used is in line 904: `for file_cat, filetypes in valid_filetypes.items():`